controllers/FrontController.php

```diff
@@ -72,4 +122,356 @@ protected function ajaxRender($value = null, $controller = null, $method = null)
             exit;
         }
     }
+
+    /**
+     * @return \Cart
+     */
+    protected function getCurrentCart()
+    {
+        return new \Cart($this->context->cart->id);
+    }
+
+    /**
+     * @param $response
+     * @param $cart
+     * @param $customer
+     * @param $isAjax
+     * @throws AdyenException
+     */
+    protected function handlePaymentsResponse($response, $cart, $customer, $isAjax)
+    {
+        $resultCode = $response['resultCode'];
+
+        $extraVars = array();
+        if (!empty($response['pspReference'])) {
+            $extraVars['transaction_id'] = $response['pspReference'];
+        }
+
+        $total = (float)$cart->getOrderTotal(true, \Cart::BOTH);
+
+        // Based on the result code start different payment flows
+        switch ($resultCode) {
+            case 'Authorised':
+                $this->module->validateOrder(
+                    $cart->id,
+                    2,
```

Comment:
Apparently you need to get this number with `Configuration::get`instead

Reply:
PS_OS_PAYMENT from the Configuration table you mean?
Yes that would be a great idea - easier to understand than just a 2 in the middle of the code ;)

Reply:
Not sure which value exactly, but yes :)